### PR TITLE
Lita dependabot security poc

### PR DESCRIPTION
**Pull request by yuenmichelle1**
lita command to query github graphql api and outputs a condensed repo -> dependabot issue count

### CODE REVIEW

File: handlers/security.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        response.reply("#{alerts.length} Alerts Total: \n #{format_alerts(repo_to_alert_count)}")
+      end
+
+      private
+
+      def filter_alerts(node_alerts, alerts, repo_to_alert_count, repo_name)
+        node_alerts.each do |alert|
+          next unless alert['dismissedAt'].nil?
+          next unless alert['fixedAt'].nil?
```
Comment (camallen) [inline handlers/security.rb:49]: no need for the nil tests here as it makes the conditions hard to read / understand if not value is nil? 
```suggestion
          next if alert['dismissedAt']
          next if alert['fixedAt']
```

File: handlers/security.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          last_repo_listed = edges[repo_count - 1]['cursor']
+          get_issues = false if repo_count < 100
+        end
+
+        response.reply("#{alerts.length} Alerts Total: \n #{format_alerts(repo_to_alert_count)}")
+      end
+
+      private
+
+      def filter_alerts(node_alerts, alerts, repo_to_alert_count, repo_name)
```
Comment (camallen) [inline handlers/security.rb:46]: maybe clarify the intention of this method, i found wasn't sure what it was filtering when i read the conditions below, perhaps something like
```suggestion
      def filter_fixed_or_dismissed_alerts(node_alerts, alerts, repo_to_alert_count, repo_name)
```

File: handlers/security.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          alerts << { repo_name => vulnerability }
+          add_alert_count(repo_to_alert_count, repo_name)
+        end
+      end
+
+      def add_alert_count(repo_to_alert_count, repo_name)
+        if repo_to_alert_count[repo_name].nil?
+          repo_to_alert_count[repo_name] = 1
+        else
+          repo_to_alert_count[repo_name] += 1
```
Comment (camallen) [inline handlers/security.rb:61]: Remove the nil test and invert the logic to help readability 
```suggestion
        if repo_to_alert_count[repo_name]
          repo_to_alert_count[repo_name] += 1
        else
          repo_to_alert_count[repo_name] = 1
```

an alternative way to write this is something like below but it may not be clearer. It does avoid the branching logic by using nil testing with || operator to assign a default or reuse the existing key value
```
repo_alert_count = repo_to_alert_count[repo_name] || 0
repo_to_alert_count[repo_name] = repo_alert_count + 1
```

File: handlers/security.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        end
+      end
+
+      def repos_to_skip
+        %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client]
```
Comment (camallen) [inline handlers/security.rb:66]: maybe move these to a constant vs a method that creates a new array each time it's used in the looping structure. Alternatively memoize the array on the method call to avoid the extra setup costs (heap space and cpu etc)
```
# memoize the value via an instance var
@repos_to_skip ||= %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client]

```

while this example is minimal in terms of heap space costs etc these things can add up if you're loop becomes bigger than you expected.

Comment (camallen) [reply handlers/security.rb:66]: also i assume these values are always case sensitive as the source data comes from the repo name in github. What happens if we rename the repo to lowercase? these will break but we haven't really changed the name. it might be useful to compare these as all lowercase to avoid this

```
# setup all lowercase repo names
@repos_to_skip ||= %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client].map(&:downcase)

# and then compare against the lowercase repo name from gh
repos_to_skip.include? repo_name.lowercase
```

File: handlers/security.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      end
+
+      def format_alerts(repo_to_alert_count)
+        formatted_alerts = "\n"
+        repo_to_alert_count.each do |repo, count|
+          formatted_alerts += "#{repo} -- #{count}\n"
+        end
+        formatted_alerts
```
Comment (camallen) [inline handlers/security.rb:74]: not blocking: An alternative way to write this could be to use the array join method. I'd remove the first new line operator and instead add this where I call this `format_alerts` method
```
  repo_to_alert_count.map { |repo, count|  "#{repo} -- #{count}" }.join('\n')  
```

File: lib/zooniverse_github.rb [hunk 6]
```diff
@@ -107,8 +107,78 @@ def update_production_migrate_tag(repo_name)
         update_tag(full_repo_name, deploy_ref)
       end
 
+      def get_dependabot_issues(last_repo_listed)
+        query = last_repo_listed.nil? ? query_without_after : query_with_after(last_repo_listed)
```
Comment (camallen) [inline lib/zooniverse_github.rb:111]: i'd remove the nil? checks here 
```suggestion
        query = last_repo_listed ? query_with_after(last_repo_listed) : query_without_after
```

**Review by yuenmichelle1 — COMMENTED** [review]
Thanks Cam! Definitely helpful comments. I'll make suggested changes.
